src/porepy/numerics/darcyEq.py

```diff
@@ -0,0 +1,175 @@
+import numpy as np
+import scipy.sparse as sps
+
+from porepy.numerics.fv import tpfa, source
+from porepy.grids.grid_bucket import GridBucket
+from porepy.params import bc, tensor
+from porepy.params.data import Parameters
+from porepy.viz.exporter import export_vtk
+
+class Darcy():
+    def __init__(self, gb, data=None, physics='flow'):
+        self.physics = physics
+        self._gb = gb
+        self._data = data
+        self.lhs, self.rhs = self.reassemble()
+        self.p = np.zeros(self.flux_disc().ndof(self.grid()))
+        self.parameters = {'file_name': physics}
+        self.parameters['folder_name'] = 'results'
+
+    def solve(self):
+        self.p = sps.linalg.spsolve(self.lhs, self.rhs)
+        return self.p
+
+    def reassemble(self):
+        """
+        reassemble matrices. This must be called between every time step to
+        update the rhs of the system.
+        """
+        lhs_flux, rhs_flux = self._discretize(self.flux_disc())
+        lhs_source, rhs_source = self._discretize(self.source_disc())
+        assert lhs_source.nnz == 0, 'Source lhs different from zero!'
+        self.lhs = lhs_flux
+        self.rhs = rhs_flux + rhs_source
+        return self.lhs, self.rhs
+
+    def source_disc(self):
+        if isinstance(self.grid(), GridBucket):
```

Comment:
This will fail if the user gives a porepy.GridBucket (short hand namespace). I believe equivalent problems can be found in several other places. We should find a more general way of doing this, perhaps by always referring to the short hand version.

Reply:
I discussed something similar with Ivar earlier this week. If the discretization of a single grid, e.g., tpfa.Tpfa took a tuple as input ( (grid, data) ) we would avoid this problem as the structure for calling mixed dim solvers and mono dim solvers would agree ( Tpfa( (g, data)), or TpfaMultiDim( gb)).